src/Features/Core/Portable/Navigation/DefaultDocumentNavigationService.cs

```diff
@@ -20,13 +20,13 @@ public Task<bool> CanNavigateToLineAndOffsetAsync(Workspace workspace, DocumentI
         public Task<bool> CanNavigateToPositionAsync(Workspace workspace, DocumentId documentId, int position, int virtualSpace, CancellationToken cancellationToken)
             => SpecializedTasks.False;
 
-        public Task<bool> TryNavigateToSpanAsync(Workspace workspace, DocumentId documentId, TextSpan textSpan, NavigationOptions options, bool allowInvalidSpan, CancellationToken cancellationToken)
-            => SpecializedTasks.False;
+        public Task<INavigableLocation?> GetLocationForSpanAsync(Workspace workspace, DocumentId documentId, TextSpan textSpan, NavigationOptions options, bool allowInvalidSpan, CancellationToken cancellationToken)
+            => SpecializedTasks.Null<INavigableLocation>();
```

Comment:
This actually seems a bit conceptually nicer in that "we can't navigate to something" is something you can find out without actually trying the operation.